orc8r/gateway/python/magma/magmad/metrics_collector.py

```diff
@@ -219,15 +218,26 @@ def _make_sync_done_func(self, service_name, chunk):
 
     def _chunk_samples(self, samples):
         # Add 1kiB fpr gRPC overhead
-        sample_size_bytes = sys.getsizeof(samples) + 1000
+        sample_size_bytes = sum(s.ByteSize() for s in samples) + 1000
+
         buckets = math.ceil(
             sample_size_bytes / self.grpc_max_msg_size_bytes,
         )
-        sample_length = len(samples)
-        chunk_size = sample_length // buckets
-
-        for i in range(0, sample_length, chunk_size):
-            yield samples[i:i + chunk_size]
+        chunk_size_bytes = sample_size_bytes // buckets
+
+        chunked_samples = []
+        chunked_samples_size = 0
+        for s in samples:
```

Comment:
If we use this logic, then we can simply iterate till `chunked_samples_size + s.ByteSize() <= chunked_samples_size + s.ByteSize() <=`, we can get rid of the buckets computation.

Reply:
We do need the buckets computation to check what is the maximum bytes size for each chunk:

```
        buckets = math.ceil(
            sample_size_bytes / self.grpc_max_msg_size_bytes,
        )
        chunk_size_bytes = sample_size_bytes // buckets
```

Reply:
Why do we need to stop at chunk_size_bytes, we can probably iterate till total number of chunks is less than the grpc msg size no?

Reply:
Ok sorry didn't get your point at first, yes basically we chunk at `self.grpc_max_msg_size_bytes`